wolfssl/wolfcrypt/port/Renesas/renesas-sce-crypt.h

```diff
@@ -0,0 +1,231 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#ifndef __RENESAS_SCE_CRYPT_H__
+#define __RENESAS_SCE_CRYPT_H__
+
+#include "r_sce.h"
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define SCE_SESSIONKEY_NONCE_SIZE           8
+#define WOLFSSL_SCE_ILLEGAL_CIPHERSUITE     -1
+
+    enum {
+    l_TLS_RSA_WITH_AES_128_CBC_SHA     = 0x2F,
+    l_TLS_RSA_WITH_AES_128_CBC_SHA256  = 0x3c,
+    l_TLS_RSA_WITH_AES_256_CBC_SHA     = 0x35,
+    l_TLS_RSA_WITH_AES_256_CBC_SHA256  = 0x3d,
+    l_TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 = 0x27,
+    l_TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256 = 0x23,
+    l_TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256  = 0x2b,
+    l_TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256    = 0x2f,
+};
+
+#define MAX_SCE_CBINDEX 5
+
+typedef struct tagUser_SCEPKCbInfo {
+    uint32_t user_key_id;
+    
+    /* out from R_SCE_TLS_ServerKeyExchangeVerify */
+    uint32_t encrypted_ephemeral_ecdh_public_key[SCE_TLS_ENCRYPTED_ECCPUBKEY_SZ];
+    /* out from R_SCE_TLS_ECC_secp256r1_EphemeralWrappedKeyPairGenerate */
+    sce_tls_p256_ecc_wrapped_key_t ecc_p256_wrapped_key;
+    uint8_t ecc_ecdh_public_key[HW_SCE_ECC_PUBLIC_KEY_BYTE_SIZE];
+    
+    uint32_t    sce_masterSecret[SCE_TLS_MASTERSECRET_SIZE/4];
+    uint8_t     sce_clientRandom[SCE_TLS_CLIENTRANDOM_SZ];
+    uint8_t     sce_serverRandom[SCE_TLS_SERVERRANDOM_SZ];
+    uint8_t     sce_cipher;
+    
+    /* installed key handling */
+    sce_aes_wrapped_key_t   sce_wrapped_key_aes256;
+    uint8_t aes256_installedkey_set:1;
+    sce_aes_wrapped_key_t   sce_wrapped_key_aes128;
+    uint8_t aes128_installedkey_set:1;
+    
+    /* flag whether encrypted ec key is set */
+    uint8_t pk_key_set:1;
+    uint8_t session_key_set:1;
+
+} User_SCEPKCbInfo;
+
+typedef struct tagSCE_PKCbInfo {
+    User_SCEPKCbInfo *user_PKCbInfo[MAX_SCE_CBINDEX];
+    uint32_t    num_session;
+} SCE_PKCbInfo;
+
+typedef struct
+{
+    uint8_t                          *encrypted_provisioning_key;
+    uint8_t                          *iv;
+    uint8_t                          *encrypted_user_tls_key;
+    uint32_t                          encrypted_user_tls_key_type;
+    sce_tls_ca_certification_public_wrapped_key_t  user_rsa2048_tls_wrappedkey;
+} sce_key_data;
+    
+struct WOLFSSL;
+struct WOLFSSL_CTX;
+struct ecc_key;
+    
+int     sce_Open();
+void    sce_Close();
+int     sce_hw_lock();
+void    sce_hw_unlock( void );
+int     sce_usable(const struct WOLFSSL *ssl);
+
+typedef struct {
+    sce_aes_wrapped_key_t   sce_wrapped_key;
+    word32                  keySize;
+    byte                    setup;
+} SCE_AES_CTX;
+
+struct Aes;
+int wc_sce_AesCbcEncrypt(struct Aes* aes, byte* out, const byte* in,
```

Comment:
Internal API should be prefixed `WOLFSSL_LOCAL`.  External/public API should be prefixed `WOLFSSL_API`. General comment to apply where needed.

Reply:
Add "wc_" to functions that we add and put WOLFSSL_LOCAL prefix when they are used for internal. Otherwise, put WOLFSSL_API for user function